src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -54,6 +57,302 @@ uint32_t SystemRequest::index = 0;
+           if (app_data[json::android][json::packageName].asString().length()>255){
+             LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED << "Package name length ["
+                          << app_data[json::android][json::packageName].asString().length()
+                 << "] exceeds max lengthCan't find package name in json file.");
+             result = false;
+           }
+         }
+       }
+
+       if (os_type.empty()) {
+         LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED << "Can't find mobile OS type in json file.");
+         result = false;
+         continue;
+       }
+
+       // languages verification
+       if (!app_data[os_type].keyExists(json::languages)) {
+         LOG4CXX_WARN(logger_, "\"languages\" not exists");
+         result = false;
+         continue;
+       }
+       if (!validateLanduages(app_data[os_type][json::languages]))
+           result = false;
+     }
+     return result;
+ }
+
+ bool validateAppIdAndAppName(const smart_objects::SmartObject& app_data) const {
+     // verify appid
+     if (!app_data.keyExists(json::appId)) {
+       LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED << "Can't find app ID in json file.");
+       return false;
+     }
+
+     // verify appid lenght
+     const std::string app_id(app_data[json::appId].asString());
+     if (app_id.length() > 100)  {
+       LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED << "An Object ID length exceeds maximum allowed ["
+                    << app_id.length() << "]>[100]");
+       return false;
+     }
+
+     // verify that appid is unique
+     if (app_ids_map_.find(app_id)!= app_ids_map_.end())     {
+       LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED
+                    << "An Object ID is not unigue [" << app_id <<  "]");
+       return false;
+     } else {
+       app_ids_map_[app_id] = 1;
+     }
+
+     // verify that app is not registered yet
+     ApplicationSharedPtr registered_app =
+         ApplicationManagerImpl::instance()->application_by_policy_id(
+             app_id);
+     if (registered_app) {
+       LOG4CXX_DEBUG(
+           logger_, QUERY_APPS_VALIDATION_FAILED <<
+           "Application with the same id: " << app_id
+                                            << " is registered already.");
+       return false;
+     }
+    // verify app name exist
+    if (!app_data.keyExists(json::name)) {
+      LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED << "Can't find app name in json file.");
+      return false;
+    }
+    // and app name length
+    const custom_str::CustomString appName(app_data[json::name].asCustomString());
+    if  (appName.length() > 100) {
+      LOG4CXX_WARN(logger_, QUERY_APPS_VALIDATION_FAILED
+                   << "Name of application exceeds maximum allowed [" << appName.length() << "]>[100].");
+      return false;
+    }
+
+     return true;
+ }
+
+
+ bool validateLanduages(const smart_objects::SmartObject& languages) const {
```

Comment:
@okoreniak Function names should start from Capital letter. Please read [Google Coding Style](https://google.github.io/styleguide/cppguide.html#Function_Names)